### PR TITLE
Docs: Build data union with core

### PR DESCRIPTION
**Pull request by juhah**
First draft of the "Build data union with core" docs article (on top of #948). Images will have to be polished later, now just simple screenshots.

---

# Build a Data Union with Core

To create a Data Union in Core, go to Core > Products and click <span className={docsStyles.streamrHighlight}>Create Product</span> and choose **Data Union** as the product type.

intro_to_products_01_desktop

Other than the sections below, the process is similar to creating a data product. Please see the section about <Link to={links.docs.products.introToProducts}>creating Data Products</Link> before proceeding with this article.

### Connect an Ethereum identity

In order to manage your data union, you need to connect your Ethereum address to your Streamr account. This is required before deploying your Data Union. Note that this won't be shown if you have already connected the current Ethereum address.

connect_eth

### Admin fee

Administrators are responsible for maintaining the Data Unions, including ensuring good data quality and removing members that are not contributing data as expected. They are incentivized to perform this work by receiving an admin fee, a fraction of incoming revenue. Members agree to it by joining the Data Union.

admin_fee

### Shared secrets

Shared secrets are only available once you have deployed your product, and provides access control for your product. It allows new members to join the Data Union without explicit approval from the Data Union admin, to streamline the on-ramp process. Users may still join the Data Union without a secret but they will have to manually accepted.

shared_secrets

### Deploying the Data Union

Creating a Data Union product starts with deploying the Data Union smart contract. A Data Union is identified by the address of its smart contract. The admin who deployes the contract then becomes the owner of the contract as well.

Once all the required fields have been given and the draft Data Union is ready to be deployed, click <span className={docsStyles.streamrHighlight}>Continue</span> to deploy the Data Union. The deploy wizard will walk you through the deploy process.

deploy_1
deploy_2
deploy_3
deploy_4

For the contract deployment transaction, you'll need a web3 wallet, and a small amount of Ethereum for gas fees. The Data Union deployment might take a moment to register once the transaction has been mined. The progress indicator will show an estimate of the remaining time. You can either close the dialog and move on or wait for the process to complete after which you will be automatically redirected.

deploy_5

### Publishing a Data Union

A minimum of one member is needed to publish the Data union. Once enough members have joined the Data Union, you can publish the Data Union by clicking <span className={docsStyles.streamrHighlight}>Publish</span> from the toolbar.

Your Data Union will be published on the Marketplace once the Ethereum blockchain has confirmed your transaction.

### Managing the Data Union

Data Unions are listed along with Data products in Core > Products. Data Unions are marked by a **Data Union** badge and a **Members** badge showing the current amount of approved members. Click the tile menu to access the Data Union stats and manage members.

product_list

The **Stats** view provides an overview of the Data Union as well as it's members and subscribers over time.

stats

The **Members** view allows you to manage join requests.

members

### CODE REVIEW

File: app/src/docs/content/dataUnions/dataUnionsInCore.mdx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      <img
+        src={SharedSecrets}
+        alt="Shared secrets"
+      />
+    </picture>
+</figure>
+
+### Deploying the Data Union
+
+Creating a Data Union product starts with deploying the Data Union smart contract. A Data Union is identified by the address of its smart contract. The admin who deployes the contract then becomes the owner of the contract as well.
```
Comment (fonty1) [inline app/src/docs/content/dataUnions/dataUnionsInCore.mdx:76]:
```suggestion
Creating a Data Union product starts with deploying the Data Union smart contract. A Data Union is identified by the address of its smart contract. The admin who deploys the contract then becomes the owner of the contract as well.
```

File: app/src/docs/content/dataUnions/dataUnionsInCore.mdx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+### Publishing a Data Union
+
+A minimum of one member is needed to publish the Data union. Once enough members have joined the Data Union, you can publish the Data Union by clicking <span className={docsStyles.streamrHighlight}>Publish</span> from the toolbar.
+
+Your Data Union will be published on the Marketplace once the Ethereum blockchain has confirmed your transaction.
+
+### Managing the Data Union
+
+Data Unions are listed along with Data products in Core > Products. Data Unions are marked by a **Data Union** badge and a **Members** badge showing the current amount of approved members. Click the tile menu to access the Data Union stats and manage members.
```
Comment (fonty1) [inline app/src/docs/content/dataUnions/dataUnionsInCore.mdx:135]:
```suggestion
Data Unions are listed along with Data products in Core > Products. Data Unions are marked by a **Data Union** badge and a **Members** badge showing the current number of approved members. Click the tile menu to access the Data Union stats and manage members.
```

**Comment by juhah** [comment]
> You might need to drop those deploy wizard screenshots, as they are a big design nono from experience 🤕 :

I left the first and the last step visible, let's adjust later. Added your suggestions in https://github.com/streamr-dev/streamr-platform/pull/950/commits/bfda70a93d0eb6b12075e9ce8110b63d0045a48f

> Also keep an eye on the /docs/dataunions hyphening on merge, I think you branched before that fix was applied on the other branch.

No worries, that's what the tests are for 🎉

**Comment by mattatgit** [comment]
@juhah Thanks for the first draft, great start. 

Overall though, as mentioned before, i think we should lean more towards MP4 screen capture videos rather than endless screenshots, because they can be difficult to read, tedious to layout & often not very useful on smaller viewports without a bunch of work to redesign the screenshots to work on those smaller screens.

Especially now we already have a workable, fairly polished app on staging to capture, i think we should go this way for all docs articles in future.

I think if we consider this approach from the start it can also inform how we write. Eg with yours above we could probably reduce it to maybe 3 MP4s + a screenshot or two, or maybe just all mp4s. Bit more work to create but not much more, now we have the app in good shape.

I'll do a pass on it & capture some video footage & see if we can simplify the layout & etc.
But yeah for any future articles let's try and focus on screen capture videos rather than screenshots.

**Pull request merged**